### PR TITLE
fix wrong dataset pass into ImageDeleter

### PR DESCRIPTION
**Pull request by noklam**
I think currently the ImageDeleter takes in a Dataset instead of a DataBunch, What I have experienced is it is deleting my training dataset while I pass in the validation idx.

See this thread.
https://forums.fast.ai/t/lesson-2-in-class-discussion/28632/740?u=nok

Reproduce:
You can reproduce this by creating a  validation set with one image only, then use the file deleter to delete it, see if it is still here.

### CODE REVIEW

**Comment by noklam** [comment]
Closed for now,  will add a more complete fix with test later
